src/west/manifest.py

```diff
@@ -1235,9 +1235,7 @@ def is_ancestor_of(self, rev1, rev2, cwd=None):
         elif rc == 1:
             return False
         else:
-            log.wrn(f'{self.name_and_path}: git failed with exit code {rc}; '
-                    f'treating as if "{rev1}" is not an ancestor of "{rev2}"')
-            return False
+            raise RuntimeError(f'unexpected git merge-base result {rc}')
```

Comment:
Previosly `west` would continue with rebasing remaining projects when this code returned `False`.
Now I assume `west` will fail, and leave the remaining projects untouched.

Thus, the overall project is in a mixed state.
Some projects are updated (rebased) according to the new manifest file, while projects not yet processed will still be pointing to the version in the `old` manifest file.

Is this the behavior we want ?
I assume that the old behavior was chosen because it allowed `west` to continue processing the remaining projects.

Reply:
The old behavior was chosen arbitrarily, to be honest. When I was looking at this code again, I decided that it wasn't very Pythonic to swallow this error code, because the git manual page says that 0 and 1 are the only possible return codes from this command. So I think only a very rare condition could cause this path to be taken, and I think raising an exception immediately is a better signal to the user that something has gone quite wrong than the current warning.

Reply:
> The old behavior was chosen arbitrarily, to be honest.

Was also my feeling, just wanted to checkup.
And I find new behavior just as good, or maybe even slightly better, as `west` will no longer hide an ultra rare error from `git`.